### PR TITLE
blockbench: Make compatible with NIXOS_OZONE_WL

### PR DESCRIPTION
**Pull request by IogaMaster**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/blockbench-electron/default.nix [hunk 1]
```diff
@@ -34,6 +34,7 @@ stdenv.mkDerivation rec {
 
   postFixup = ''
     makeWrapper ${electron_25}/bin/electron $out/bin/${pname} \
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--enable-features=UseOzonePlatform --ozone-platform=wayland}}" \
```
Comment (eclairevoyant) [inline pkgs/applications/graphics/blockbench-electron/default.nix:37]:
```suggestion
      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

**Comment by ckiee** [comment]
Out of scope for me since I don't use wayland currently.

**Comment by IogaMaster** [comment]
s have someone else that can test for me

**Comment by ckiee** [comment]
You can go ahead and apply&rebase @eclairevoyant's review, it seems to be the current consensus on what flags to pass[^1]. I'll test on a nested sway once we're there.

[^1]: [arch wiki](https://wiki.archlinux.org/title/Wayland#Command_line_flags) and it's more popular in nixpkgs.

**Comment by IogaMaster** [comment]
I though so, I copied from another package that used this instead

**Comment by TomaSajt** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/279795

**Pull request closed without merging**